src/analyze/analyze.c

```diff
@@ -1136,28 +1153,37 @@ static int graph_one_property(sd_bus *bus, const UnitInfo *u, const char* prop,
 static int graph_one(sd_bus *bus, const UnitInfo *u, char *patterns[], char *from_patterns[], char *to_patterns[]) {
         int r;
 
+        typedef struct ArgDotDep {
+                enum dot arg_dot;
+                UnitDependency dependency;
+        } ArgDotDep;
+
+        static const ArgDotDep arg_dot_deps[] = {
+                { DEP_ORDER,    UNIT_AFTER      },
+                { DEP_REQUIRE,  UNIT_REQUIRES   },
+                { DEP_REQUIRE,  UNIT_REQUISITE  },
+                { DEP_REQUIRE,  UNIT_WANTS      },
+                { DEP_REQUIRE,  UNIT_BINDS_TO   },
+                { DEP_REQUIRE,  UNIT_PART_OF    },
+                { DEP_REQUIRE,  UNIT_CONFLICTS  },
+                { DEP_FAILURE,  UNIT_ON_FAILURE },
+                { DEP_TRIGGERS, UNIT_TRIGGERS   },
+                {}
+        };
+
         assert(bus);
         assert(u);
 
-        if (IN_SET(arg_dot, DEP_ORDER, DEP_ALL)) {
-                r = graph_one_property(bus, u, "After", "green", patterns, from_patterns, to_patterns);
-                if (r < 0)
-                        return r;
-        }
-
-        if (IN_SET(arg_dot, DEP_REQUIRE, DEP_ALL)) {
-                r = graph_one_property(bus, u, "Requires", "black", patterns, from_patterns, to_patterns);
-                if (r < 0)
-                        return r;
-                r = graph_one_property(bus, u, "Requisite", "darkblue", patterns, from_patterns, to_patterns);
-                if (r < 0)
-                        return r;
-                r = graph_one_property(bus, u, "Wants", "grey66", patterns, from_patterns, to_patterns);
-                if (r < 0)
-                        return r;
-                r = graph_one_property(bus, u, "Conflicts", "red", patterns, from_patterns, to_patterns);
-                if (r < 0)
-                        return r;
+        for (int i = 0; arg_dot_deps[i].arg_dot; i++) {
```

Comment:
int → size_t (after all this is a memory object we iterate through here)

Also, for local tables like this consider iterating from 0 to ELEMENTSOF(…). This is nice, as it means we don't have to have the special, trailing entry

Reply:
done